### PR TITLE
Changelog

### PR DESCRIPTION
**Pull request by motss**
## Changes

1. To improve workflow in updating changelogs

### CODE REVIEW

**Comment by motss** [comment]
Closing this PR after the experiments:

1. Tried to always update changelogs after each commit at `post-commit` level by updating all changelogs and commiting all updated changelogs then force push a new commit. However this causes the reference to the previous commit to be no longer existed and it does not update the reference to the new commit in the changelog.
2. Tried to always update changelogs before at `pre-commit` level so that changes from previous commit will be included at the current commit. However this raises another problem is that PRs will inevitably cause the changelogs to include all changes even when the PR is squashed and merged into the main branch. This worsen for a project which involves multiple collaborators.

TLDR; The best way to update changelogs is when all the changes are ready to be released in a new tag. The only caveat with this approach is that not all updates are always included as unreleased changes in the changelog but this is not a big deal as nightly builds are always risky to be used in a daily basis. Changes are not always finalized before it is ready to be released.
